spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/SleuthKafkaStreamsConfiguration.java

```diff
@@ -66,8 +67,8 @@ static KafkaStreamsTracing kafkaStreamsTracing(Tracing tracing) {
 
 	@Bean
 	static KafkaStreamsBuilderFactoryBeanPostProcessor kafkaStreamsBuilderFactoryBeanPostProcessor(
-			KafkaStreamsTracing kafkaStreamsTracing) {
-		return new KafkaStreamsBuilderFactoryBeanPostProcessor(kafkaStreamsTracing);
+			ObjectProvider<KafkaStreamsTracing> objectProvider) {
```

Comment:
I think this method deserves a comment about why ObjectProvider is needed (not saying it is wrong)